packages/components/src/2016-12-01/button-state.js

```diff
@@ -14,28 +14,43 @@ See the License for the specific language governing permissions and
 limitations under the License.
 
 */
+function prevState(state) {
```

Comment:
I removed the prevState function, and separated element.state from element.viewState so we can distinguish configuration from appearance much more cleanly.